icepyx/core/read.py

```diff
@@ -395,35 +437,65 @@ def _add_vars_to_ds(is2ds, ds, grp_path, wanted_groups_tiered, wanted_dict):
                     import random
 
-                    is2ds["gran_idx"] = [random.randint(900000, 999999)]
+                    is2ds["gran_idx"] = [random.randint(800000, 899998)]
                     warnings.warn("Your granule index is made up of random values.")
                     # You must include the orbit/cycle_number and orbit/rgt variables to generate
             except KeyError:
-                pass
+                # Added this when dealing with ATL11 - need to see if it breaks with other datasets
+                is2ds["gran_idx"] = [np.nanmax(is2ds["gran_idx"]) - 1]
+                # pass
 
             if hasattr(is2ds, "data_start_utc"):
                 is2ds = _make_np_datetime(is2ds, "data_start_utc")
                 is2ds = _make_np_datetime(is2ds, "data_end_utc")
 
         else:
-            import re
+            track_str, spot_dim_name = _get_track_type_str(grp_path)
 
-            gt_str = re.match(r"gt[1-3]['r','l']", grp_path).group()
-            spot = is2ref.gt2spot(gt_str, is2ds.sc_orient.values[0])
-            # add a test for the new function (called here)!
+            # get the spot number if relevant
+            if spot_dim_name == "spot":
+                spot = is2ref.gt2spot(track_str, is2ds.sc_orient.values[0])
+            else:
+                spot = track_str
 
             grp_spec_vars = [
                 k
                 for k, v in wanted_dict.items()
                 if any(f"{grp_path}/{k}" in x for x in v)
             ]
 
+            # NEXT TODO: handle the case where it's the second time through a 2d delta_time...
+
+            # handle delta_times with 1 or more dimensions
+            idx_range = range(0, len(ds.delta_time.data))
+            # if hasattr(is2ds, "photon_idx"):
+
+            #     # if is2ds already has a 2d photon idx/delta time AND the current delta time does too
+            #     if np.ndim(ds.delta_time.data) > 1: # and np.ndim(is2ds.photon_idx) > 1:
+            #         # repeat the range the number of times needed, then transpose to match the shape of the existing photon_idx
+            #         photon_ids = (
+            #         np.broadcast_to([*idx_range], (np.shape(ds.delta_time)[1], np.shape(ds.delta_time)[0])).transpose()
+            #         + np.full_like(
+            #             ds.delta_time, np.max(is2ds.photon_idx), dtype="int64"
+            #         )
+            #         + 1
+            #     )
+            #     # the original case, where delta_time is 2d but the existing photon_idx is 1d
+            #     else:
+            #         photon_ids = (
+            #             range(0, len(ds.delta_time.data))
+            #             + np.full_like(
+            #                 ds.delta_time, np.max(is2ds.photon_idx), dtype="int64"
+            #             )
+            #             + 1
+            #         )
+            # else:
+            #     photon_ids = range(0, len(ds.delta_time.data))
+
```

Comment:
Can these commented lines be removed, or did you want to keep them for something?

Reply:
Yup. They were leftover that I missed during cleanup.